File: lectures/14/Books/app/src/main/java/ro/ubbcluj/cs/books/BookApp.java
package ro.ubbcluj.cs.books;

import android.app.Application;

import io.realm.Realm;
import io.realm.RealmConfiguration;
import timber.log.Timber;

public class BookApp extends Application {
  @Override
  public void onCreate() {
    super.onCreate();
    Timber.plant(new Timber.DebugTree());
    Realm.init(this);
    RealmConfiguration config = new RealmConfiguration.Builder().build();
    Realm.deleteRealm(config);
    Realm.setDefaultConfiguration(config);
  }
}

File: lectures/12/Books/app/src/main/java/ro/ubbcluj/cs/books/BookApp.kt
package ro.ubbcluj.cs.books

import android.app.Application
import androidx.room.Room

import ro.ubbcluj.cs.books.db.BookDatabase

class BookApp : Application() {
  lateinit var db: BookDatabase

  override fun onCreate() {
    super.onCreate()
    db = Room.databaseBuilder(applicationContext,
        BookDatabase::class.java, "database-name").build()
  }
}
